### PR TITLE
ExperimentalScanner fixes

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Some fixes for the experimental scanner, see the commit messages for details.

### CODE REVIEW

File: scanner/src/main/kotlin/experimental/NestedProvenanceScanResult.kt [hunk 1]
```diff
@@ -167,10 +167,15 @@ data class NestedProvenanceScanResult(
...
             val provenancePath = getPath(it)
-            provenancePath.isEmpty() || path.startsWith("$provenancePath/")
+            provenancePath.isEmpty() || provenancePath == path || provenancePath.startsWith("$path/") || path.startsWith("$provenancePath/")
         }.associateWith { provenance ->
             val provenancePath = getPath(provenance)
-            path.removePrefix("$provenancePath/")
+
+            when {
+                provenancePath.isEmpty() -> path
+                path.startsWith("$provenancePath/") -> path.removePrefix("$provenancePath/")
```
Comment (sschuberth) [inline scanner/src/main/kotlin/experimental/NestedProvenanceScanResult.kt:171]: Just checking: We don't need the the same logic for `"$path/"`, right?

**Review by sschuberth — DISMISSED** [review]

File: scanner/src/main/kotlin/experimental/NestedProvenanceScanResult.kt (hunk 1 above)
Comment (mnonnenmacher) [reply scanner/src/main/kotlin/experimental/NestedProvenanceScanResult.kt:171]: No, above the check is reciprocal to figure out if the subrepository is on the same branch as the provided filter path, but for getting the relative path within the subrepository we don't need it.

Comment (sschuberth) [reply scanner/src/main/kotlin/experimental/NestedProvenanceScanResult.kt:171]: Thanks for confirming. Would it make sense to put that sentence of yours into a code comment above the `when`?

Comment (mnonnenmacher) [reply scanner/src/main/kotlin/experimental/NestedProvenanceScanResult.kt:174]: I have added a few explanatory code comments.

**Review by sschuberth — APPROVED** [review]

**Comment by sschuberth** [comment]
FYI @mnonnenmacher, we currently have issues with the Markdown check / GitHub's checkout action, so that test is expected to fail.

**Pull request merged**